modules/developing/pages/astream-functions.adoc

```diff
@@ -229,229 +185,253 @@ Result::
 [INFO] Finished at: 2023-05-16T16:19:05-04:00
 [INFO] ------------------------------------------------------------------------
 ----
---
 ====
+--
 
-. Create a deployment configuration file. In this example we'll call this file “func-create-config.yaml”.
-This file will be passed to the pulsar-admin create function command. +
-The contents of the YAML file should be:
+////
+TODO:
+Go functions::
 +
-[source,yaml]
+--
+Must have 4 steps to maintain numbering.
+--
+////
+======
+
+[start=5]
+. If you haven't done so already, xref:configure-pulsar-env.adoc[set up your environment for the Pulsar binaries].
+
+. Create a deployment configuration YAML file that defines the function metadata and associated topics:
++
+.func-create-config.yaml
+[source,yaml,subs="+quotes"]
 ----
-jar: </absolute/path/to/my/testpythonfunction.jar>
-className: com.example.pulsar.ExclamationFunction
+py: **PATH_TO_FUNCTION_ARCHIVE**
+className: **FILE_NAME**.**CLASS_NAME**
 parallelism: 1
 inputs:
-  - persistent://mytenant/n0/t1
-output: persistent://mytenant/ns/t2
+ - persistent://**TENANT_NAME**/**NAMESPACE_NAME**/**TOPIC_NAME**
+output: persistent://**TENANT_NAME**/**NAMESPACE_NAME**/**TOPIC_NAME**
 autoAck: true
-tenant: mytenant
-namespace: ns0
-name: testjarfunction
+tenant: **TENANT_NAME**
+namespace: **NAMESPACE_NAME**
+name: **DISPLAY_NAME**
 logTopic:
 userConfig:
-  logging_level: ERROR
+ logging_level: ERROR
 ----
 +
-[NOTE]
+Replace the following:
++
+* `**PATH_TO_FUNCTION_ARCHIVE**`: The path to the function archive.
+
+* `**FILE_NAME**.**CLASS_NAME**`: The class to execute.
+An archive can contain multiple classes, but only one is used per deployment.
++
+** For Python scripts, the `className` is the Python filename (without the extension) and the class to execute, such as `pythonfunc.ExclamationFunction`.
+If there is no class in the file, the `className` is the filename without the extension, such as `pythonfunc`.
+** For Java scripts, the `className` is the path and the class to execute, such as `com.example.pulsar.ExclamationFunction`.
+
+* `**TENANT_NAME**/**NAMESPACE_NAME**/**TOPIC_NAME**`: Define the tenant, namespace, and topic for both `input` (incoming messages passed to the function) and `output` (the results of the function).
++
+[IMPORTANT]
 ====
-Astra Streaming requires the “inputs” topic to have a message schema defined before deploying the function. Otherwise, deployment errors may occur. Use the {astra_ui} to define the message schema for a topic.
+To avoid errors, make sure the `inputs` topic (declared in your configuration YAML file) has a defined message schema before you deploy the function.
```

Comment:
```suggestion
To avoid errors, make sure the `inputs` topic (declared in your configuration YAML file) has a defined message schema before deploying the function.
```